### PR TITLE
chore(RHINENG-9807): use api v2 to fetch playbook preview

### PR DESCRIPTION
**Pull request by mkholjuraev**
# Description

Associated Jira ticket: # [RHINENG-9807](https://issues.redhat.com/browse/RHINENG-9807)

Changes the API v1 to API v2 to fetch playbook preview.

# How to test the PR

Please include steps to test your PR.

# Before the change

# After the change

# Dependent work link

# Checklist:

- [ ] The commit message has the Jira ticket linked
- [ ] PR has a short description
- [ ] Screenshots before and after the change are added
- [ ] Tests for the changes have been added
- [ ] README.md is updated if necessary
- [ ] Needs additional dependent work

### CODE REVIEW

File: src/api/index.js [hunk 1]
```diff
@@ -37,7 +36,7 @@ export const useGetPlaybookPreview = (data) => {
         const previewData = dataTransformer(data);
         const playbookPreview =
           previewData &&
-          (await axios.post(`${V1_API_BASE}/states/preview`, {
+          (await axios.post(`${CONNECTOR_API_BASE}/states/preview`, {
```
Comment (r14chandra) [inline src/api/index.js]: @mkholjuraev there is no `v2/states/preview` endpoint in v2 API.
You can use this new endpoint [/playbooks](https://github.com/RedHatInsights/config-manager/blob/master/internal/http/v2/openapi.json#L204) of v2 with `profile_id` param i.e `/v2/playbooks?profile_id=%v`

**Review by r14chandra — CHANGES_REQUESTED** [review]

File: src/api/index.js (hunk 1 above)
Comment (mkholjuraev) [reply src/api/index.js]: @r14chandra aha, thank you for pointing this out. Can you give more context on what are the differences between the old endpoint and the new one? Unfortunately, the OpenApi docs do not really show this.

Comment (r14chandra) [reply src/api/index.js]: Both are the same. Earlier we used to send a state map in v1/states/preview endpoint -> `{"compliance_openscap":"enabled","insights":"enabled","remediations":"enabled"}` but using v2/playbooks endpoint we send `profile_id` param.
`v2` API was introduced to resolve race condition in `v1` API - [ESSNTL-3301](https://issues.redhat.com/browse/ESSNTL-3301). Let me know if this helps.

Comment (r14chandra) [reply src/api/index.js]: As of now, the UI gets a state map from the Model. But now we need to pass `profile_id` of the current config state. I think here we can use `v2/profiles/current` to get the currently applied config and from there we can get the profile ID. What do you think?

Comment (mkholjuraev) [reply src/api/index.js]: Yes, that should be doable. Thank you!

Comment (mkholjuraev) [reply src/api/index.js]: @r14chandra now, everything should be fine. PTAL!

Comment (mkholjuraev) [reply src/api/index.js]: At the beginning, I did think the v1 just needed to be changed to v2 which was crazy :)))

**Review by r14chandra — APPROVED** [review]
Thanks for working on it, @mkholjuraev! The changes look good. Tested locally! 🚀

**Pull request merged**